src/test/java/org/datadog/jmxfetch/TestGCMetrics.java

```diff
@@ -0,0 +1,307 @@
+                "              domain: Bohnanza"
+            );
+            this.app.doIteration();
+            final List<Map<String, Object>> metrics = ((ConsoleReporter) this.appConfig.getReporter()).getMetrics();
+            assertThat(metrics, hasSize(1));
+        }
+    }
+
+    @Test
+    public void testDefaultOldGC() throws IOException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(RMI_PORT, CONTROL_PORT,
+            SUPERVISOR_PORT)) {
+            server.start();
+            final String ipAddress = server.getIp();
+            this.initApplicationWithYamlLines("init_config:",
+                "  is_jmx: true",
+                "",
+                "instances:",
+                "    -   name: jmxint_container",
+                "        host: " + ipAddress,
+                "        collect_default_jvm_metrics: true",
+                "        max_returned_metrics: 300000",
+                "        port: " + RMI_PORT);
+            this.app.doIteration();
+            final List<Map<String, Object>> actualMetrics = ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+            List<String> gcGenerations = Arrays.asList(
+                "G1 Old Generation",
+                "G1 Young Generation");
+            assertGCMetric(actualMetrics, "jvm.gc.cms.count", gcGenerations);
+            assertGCMetric(actualMetrics, "jvm.gc.parnew.time", gcGenerations);
+        }
+    }
+
+    @Test
+    public void testDefaultNewGCMetricsUseParallelGC() throws IOException, InterruptedException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(
+                MisbehavingJMXServer.JDK_11,
+                "-XX:+UseParallelGC",
+                RMI_PORT,
+                CONTROL_PORT,
+                SUPERVISOR_PORT)) {
+            server.start();
+            final String ipAddress = server.getIp();
+            this.initApplicationWithYamlLines("init_config:",
+                "  is_jmx: true",
+                "  new_gc_metrics: true",
+                "",
+                "instances:",
+                "    -   name: jmxint_container",
+                "        host: " + ipAddress,
+                "        collect_default_jvm_metrics: true",
+                "        max_returned_metrics: 300000",
+                "        port: " + RMI_PORT);
+            // Run one iteration first
+            this.app.doIteration();
+            // And then pull get the metrics or else reporter does not have correct number of metrics
+            ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+
+            // Actual iteration we care about
+            this.app.doIteration();
+            final List<Map<String, Object>> actualMetrics = ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+            assertThat(actualMetrics, hasSize(13));
+            final List<String> gcYoungGenerations = Collections.singletonList(
+                    "PS Scavenge");
```

Comment:
Implementation of all 3 of these GC tests look very similar, any value in refactoring them to share an implementation? (I'll leave it up to you, no strong opinion from me)